File: go/src/euler/002.go
//By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

package main

import "fmt"

func main() {
	fib, last, sum := 1, 1, 0

	for fib < 4000000 {
		fib, last = fib+last, fib
		if fib%2 == 0 {
			sum += fib
		}
	}
	fmt.Println(sum)
}

File: rust/src/002.rs
fn main() {
    let mut sum = 0;
    let mut last = 1;
    let mut fib = 1;

    while fib < 4000000 {
        if fib % 2 == 0 {
            sum = sum + fib;
        }
        let temp = fib;
        fib = fib + last;
        last = temp;
    }

    println!("{}", sum);
}